packages/arb-token-bridge-ui/src/components/TransactionHistory/TransactionsTableDetailsTeleporterSteps.tsx

```diff
@@ -0,0 +1,182 @@
+import { twMerge } from 'tailwind-merge'
+import {
+  ArrowTopRightOnSquareIcon,
+  CheckCircleIcon
+} from '@heroicons/react/24/outline'
+import { MergedTransaction } from '../../state/app/state'
+import { getExplorerUrl, getNetworkName, isNetwork } from '../../util/networks'
+import {
+  firstRetryableLegRequiresRedeem,
+  getChainIdForRedeemingRetryable,
+  l1L2RetryableRequiresRedeem,
+  l2ForwarderRetryableRequiresRedeem,
+  secondRetryableLegForTeleportRequiresRedeem
+} from '../../util/RetryableUtils'
+import { TransactionsTableRowAction } from './TransactionsTableRowAction'
+import { ExternalLink } from '../common/ExternalLink'
+import { Step } from './TransactionsTableDetailsSteps'
+import { DepositCountdown } from '../common/DepositCountdown'
+
+const TeleportMiddleStepFailureExplanationNote = ({
+  tx
+}: {
+  tx: MergedTransaction
+}) => {
+  return (
+    <div className="mt-2">
+      <div
+        className={twMerge(
+          'flex items-center gap-2',
+          l1L2RetryableRequiresRedeem(tx) && 'text-white/60'
+        )}
+      >
+        {!l1L2RetryableRequiresRedeem(tx) ? (
+          <CheckCircleIcon
+            height={15}
+            className="h-[15px] w-[15px] shrink-0 text-green-400"
+          />
+        ) : (
+          <div className="m-[1px] h-[13px] w-[13px] shrink-0 rounded-full border border-white/60" />
+        )}
+        Funds deposited to {getNetworkName(getChainIdForRedeemingRetryable(tx))}
+        .{' '}
+      </div>
+      <div
+        className={twMerge(
+          'mt-1 flex items-center gap-2',
+          l2ForwarderRetryableRequiresRedeem(tx) && 'text-white/60'
+        )}
+      >
+        {!l2ForwarderRetryableRequiresRedeem(tx) ? (
+          <CheckCircleIcon
+            height={15}
+            className="h-[15px] w-[15px] shrink-0 text-green-400"
+          />
+        ) : (
+          <div className="m-[1px] h-[13px] w-[13px] shrink-0 rounded-full border border-white/60" />
+        )}
+        Funds sent to {getNetworkName(tx.destinationChainId)}.
+      </div>
+    </div>
+  )
+}
+
+export const TransactionsTableDetailsTeleporterSteps = ({
+  tx,
+  address
+}: {
+  tx: MergedTransaction
+  address: Address | undefined
+}) => {
+  const { isTestnet: isTestnetTx } = isNetwork(tx.childChainId)
+
+  const l2TxID = tx.l1ToL2MsgData?.l2TxID
+  const isFirstRetryableLegSucceeded =
+    typeof l2TxID !== 'undefined' &&
+    typeof tx.l2ToL3MsgData?.l2ForwarderRetryableTxID === 'undefined'
+  const l2ChainId = tx.l2ToL3MsgData?.l2ChainId
+  const isFirstRetryableLegFailed = firstRetryableLegRequiresRedeem(tx)
+  const l2ForwarderRequiresRedeem = l2ForwarderRetryableRequiresRedeem(tx)
+
+  const isFirstRetryableLegResolved =
+    isFirstRetryableLegSucceeded || isFirstRetryableLegFailed
+
+  const isSecondRetryableLegResolved =
+    secondRetryableLegForTeleportRequiresRedeem(tx) ||
+    typeof tx.l2ToL3MsgData?.l3TxID !== 'undefined'
+
+  const firstTransactionText = useMemo(() => {
```

Comment:
It is safe to use `useMemo` for JSX. It is also a part of example of official react docs.
https://legacy.reactjs.org/docs/hooks-faq.html#how-to-memoize-calculations
https://stackoverflow.com/questions/60453845/is-it-safe-to-usememo-for-jsx

Re: moving this to a separate component, I would prefer the `firstTransactionText` to be scoped within the `TransactionsTableDetailsTeleporterSteps` since we don't see a possibility of using it anywhere else.